plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java

```diff
@@ -168,11 +175,27 @@ public void performAction(IncrementalTaskInputs inputs) throws Exception {
 				.steps(steps)
 				.exceptionPolicy(exceptionPolicy)
 				.build()) {
+			// determine if a list of files has been passed in
+			Predicate<File> shouldInclude;
+			Project project = getProject();
+			if (project.hasProperty(FILES_PROPERTY) && project.property(FILES_PROPERTY) instanceof String) {
+				Object rawIncludePatterns = project.property(FILES_PROPERTY);
+				final String[] includePatterns = ((String) rawIncludePatterns).split(",");
+				final List<Pattern> compiledIncludePatterns = Arrays.stream(includePatterns)
+						.map(Pattern::compile)
+						.collect(Collectors.toList());
+				shouldInclude = file -> compiledIncludePatterns
+						.stream()
+						.anyMatch(filePattern -> filePattern.matcher(file.getAbsolutePath())
+								.matches());
+			} else {
+				shouldInclude = file -> true;
+			}
 			// find the outOfDate files
 			List<File> outOfDate = new ArrayList<>();
 			inputs.outOfDate(inputDetails -> {
 				File file = inputDetails.getFile();
-				if (file.isFile() && !file.equals(getCacheFile())) {
+				if (file.isFile() && !file.equals(getCacheFile()) && shouldInclude.test(file)) {
```

Comment:
Your predicate has no IO and is thus probably the fastest one, so I would put it first.

Reply:
updated